### PR TITLE
allow mutliple listeners & auth

### PR DESCRIPTION
**Pull request by emoruzzi**

### CODE REVIEW

**Comment by sleighzy** [comment]
Hi @emoruzzi . Thanks for the contribution, much appreciated. Sorry about the delay on this, I'm currently on vacation. Will run through this in greater detail when I return and get back to with any feedback or merge. Thanks.

File: defaults/main.yaml [hunk 1]
```diff
@@ -39,17 +39,16 @@ kafka_background_threads: 10
 #   EXAMPLE:
 #     listeners = PLAINTEXT://your.host.name:9092
 #listeners=PLAINTEXT://:9092
-kafka_listener_protocol: PLAINTEXT
-kafka_listener_hostname: localhost
-kafka_listener_port: 9092
+kafka_listener:
```
Comment (sleighzy) [inline defaults/main.yaml]: Rename to `kafka_listeners` as this is a list.

File: templates/server.properties.j2 [hunk 2]
```diff
@@ -166,3 +168,28 @@ log.cleaner.backoff.ms={{ kafka_log_cleaner_backoff_ms }}
 # We override this to 0 here as it makes for a better out-of-the-box experience for development and testing.
 # However, in production environments the default value of 3 seconds is more suitable as this will help to avoid unnecessary, and potentially expensive, rebalances during application startup.
 group.initial.rebalance.delay.ms={{ kafka_group_initial_rebalance_delay_ms }}
+
+{% if kafka_auth_listener is defined %}
+############################# AUTHENTICATION #############################
+  {% for type in kafka_auth_listener %}
+{{ type.name }} required \
+  {% for user in type.users %}
```
Comment (sleighzy) [inline templates/server.properties.j2:176]: Can indent this as is nested loop.

File: defaults/main.yaml [hunk 3]
```diff
@@ -169,3 +168,25 @@ kafka_producer_compression_type: none
 
 kafka_consumer_group_id: kafka-consumer-group
 kafka_consumer_bootstrap_servers: "localhost:9092"
+
+# Example authentication
+############################# AUTHENTICATION #############################
+# kafka_auth_listener:
```
Comment (sleighzy) [inline defaults/main.yaml]: Can you flip the order of this so that it is `kafka_listener_auth`

**Review by sleighzy — CHANGES_REQUESTED** [review]
Looking good, couple of minor changes.

File: defaults/main.yaml (hunk 1 above)
Comment (emoruzzi) [reply defaults/main.yaml]: updated

File: templates/server.properties.j2 (hunk 2 above)
Comment (emoruzzi) [reply templates/server.properties.j2:176]: updated

File: defaults/main.yaml (hunk 3 above)
Comment (emoruzzi) [reply defaults/main.yaml]: updated here and other places

File: templates/server.properties.j2 [hunk 4]
```diff
@@ -44,13 +44,15 @@ background.threads={{ kafka_background_threads }}
 #   EXAMPLE:
 #     listeners = PLAINTEXT://your.host.name:9092
 #listeners=PLAINTEXT://:9092
-listeners={{ kafka_listener_protocol }}://{{ kafka_listener_hostname }}:{{ kafka_listener_port }}
+listeners={{ kafka_listener | join(",")}}
```
Comment (sleighzy) [inline templates/server.properties.j2]: `kafka_listener` -> `kafka_listeners`

**Review by sleighzy — CHANGES_REQUESTED** [review]
Just needs the update to `kafka_listeners` and then should be good to go.

**Comment by sleighzy** [comment]
Hi @emoruzzi . Are you able to make the outstanding change to have this PR completed? Would be good to see this as part of the role so I am appreciating your efforts.

File: templates/server.properties.j2 (hunk 4 above)
Comment (emoruzzi) [reply templates/server.properties.j2]: updated

**Comment by emoruzzi** [comment]
Fixed the issues - sorry didn't get the last email. Thanks for checking back in.

**Review by sleighzy — APPROVED** [review]

**Pull request merged**